File: Ridy-taxi-app/ios/common/Models/Objects/Coupon.swift
//
//  Coupon.swift
//
//  Created by Minimalistic apps on 11/15/18
//  Copyright (c) . All rights reserved.
//

import Foundation

public final class Coupon: Codable {
    
    // MARK: Properties
    
    public var id: Int?
    public var title: String?
    public var description: String?
    public var startTimestamp: Double?
    public var expirationTimestamp: Double?
    public var isEnabled: Bool?
    public var manyTimesUserCanUse: Int?
    public var manyUsersCanUse: Int?
    public var isFirstTravelOnly: Bool?
    public var maximumCost: Int?
    public var code: String?
    public var creditGift: Int?
    public var discountFlat: Int?
    public var minimumCost: Int?
    public var discountPercent: Int?
}

File: Ridy-taxi-app/android/common/src/main/java/com/innomalist/taxi/common/models/Coupon.kt
package com.innomalist.taxi.common.models

import com.squareup.moshi.JsonClass
import java.io.Serializable
import java.sql.Timestamp
import java.util.*

@JsonClass(generateAdapter = true)
class Coupon : Serializable {
    var isEnabled: Boolean = true
    var manyUsersCanUse = 0
    var manyTimesUserCanUse = 0
    var flatDiscount: Double = 0.0
    var code: String? = null
    var description: String? = null
    var id = 0
    var title: String? = null
    var startTimestamp: Long = 0
    var expirationTimestamp: Long = 0
    var discountPercent = 0
    var isFirstTravelOnly: Boolean = false
    var daysLeft = 0
        get() {
            field = ((expirationTimestamp - Date().time) / (1000 * 60 * 60 * 24)).toInt()
            return field
        }

}